be/src/exprs/agg/histogram.h

```diff
@@ -0,0 +1,187 @@
+            Bucket<T> bucket(v, v, 1, 1);
+            this->data(state).buckets.push_back(bucket);
+        } else {
+            Bucket<T>* lastBucket = &this->data(state).buckets.back();
+
+            if (lastBucket->upper == v) {
+                lastBucket->count++;
+                lastBucket->count_in_bucket++;
+                lastBucket->upper_repeats++;
+            } else {
+                if (lastBucket->count_in_bucket >= bucket_size) {
+                    Bucket<T> bucket(v, v, lastBucket->count + 1, 1);
+                    this->data(state).buckets.push_back(bucket);
+                } else {
+                    lastBucket->upper = v;
+                    lastBucket->count++;
+                    lastBucket->count_in_bucket++;
+                    lastBucket->upper_repeats = 1;
+                }
+            }
+        }
+    }
+
+    void update_batch_single_state(FunctionContext* ctx, AggDataPtr __restrict state, const Column** columns,
+                                   int64_t peer_group_start, int64_t peer_group_end, int64_t frame_start,
+                                   int64_t frame_end) const override {
+        //Histogram aggregation function only support one stage Agg
+        DCHECK(false);
+    }
+
+    void merge(FunctionContext* ctx, const Column* column, AggDataPtr __restrict state, size_t row_num) const override {
+        //Histogram aggregation function only support one stage Agg
+        DCHECK(false);
+    }
+
+    void serialize_to_column(FunctionContext* ctx __attribute__((unused)), ConstAggDataPtr __restrict state,
+                             Column* to) const override {
+        //Histogram aggregation function only support one stage Agg
+        DCHECK(false);
+    }
+
+    void convert_to_serialize_format(FunctionContext* ctx, const Columns& src, size_t chunk_size,
+                                     ColumnPtr* dst) const override {
+        //Histogram aggregation function only support one stage Agg
+        DCHECK(false);
+    }
+
+    std::string toBucketJson(std::string lower, std::string upper, size_t count, size_t upper_repeats,
+                             double sample_ratio) const {
+        std::string bucket_json_array = "[\"" + lower + "\",\"" + upper + "\",\"" +
+                                        std::to_string((size_t)(count * sample_ratio)) + "\",\"" +
+                                        std::to_string((size_t)(upper_repeats * sample_ratio)) + "\"]";
+        return bucket_json_array;
+    }
+
+    void finalize_to_column(FunctionContext* ctx __attribute__((unused)), ConstAggDataPtr __restrict state,
+                            Column* to) const override {
+        std::string histogram_json = "{ \"buckets\" : [";
+        if constexpr (pt_is_largeint<PT>) {
+            for (int i = 0; i < this->data(state).buckets.size(); ++i) {
+                histogram_json +=
+                        toBucketJson(LargeIntValue::to_string(this->data(state).buckets[i].lower),
```

Comment:
Should these if-else be moved to a general `to_string` implementation? Since that's the only difference?